### DISCUSSION TITLE
Printable Layouts for Keymap

### DISCUSSION THREAD

**Discussion by amerikiwi93** [manna-harbour/miryoku#71] [Q&A] [answered]
I would like to print layouts for the keymaps. Currently the main layout has great background for the keys. Is it possible to get a copy or way for me to edit the current version so I can edit it myself?

**Comment by manna-harbour** [comment, marked as answer]
I will be making diagrams suitable for printing in future (#72).

To manually modify the current diagrams or to obtain a higher resolution image:

1. Find the JSON source under https://github.com/manna-harbour/miryoku/tree/master/data/layers
2. Download the JSON source, e.g https://raw.githubusercontent.com/manna-harbour/miryoku/master/data/layers/miryoku-kle-reference.json
3. Open http://www.keyboard-layout-editor.com/
4. Select the `</> Raw data` tab
5. Press the `Upload JSON` button
6. Upload the JSON source
7. Edit (See `Help & keyboard shortcuts`)
8. Press the `Download` button
9. Select `Download PNG` and `Download JSON` 
10. For a higher resolution image, zoom and screenshot, or print to PDF and edit

**Reply by Lenbok** [reply]
There are json files for the main layers, but how can we get versions for flipped (without inverted t)?

**Reply by manna-harbour** [reply]
@Lenbok Files are only available for default and for flip with inverted-T. For any other combination of alternative layout options you'd have to edit the files manually as described. If you do that it's probably easier to move the keys around and then change the Base layer legends and homing indicators rather than editing all the sub layer legends. You can refer to the source tables in Miryoku Babel for the positions.

Also, betas of custom generated diagrams are now available, see https://github.com/manna-harbour/miryoku/issues/72#issuecomment-1154588220.

**Reply by mptrsen** [reply]
This doesn't work (anymore?), the downloaded SVG file from http://www.keyboard-layout-editor.com/ lacks all legends. The keys are just blank. Downloaded PNG/JPG images are low resolution unsuitable for printing.

**Reply by manna-harbour** [reply]
@mptrsen For a higher resolution image, zoom and screenshot, print to PDF and edit, or see https://github.com/manna-harbour/miryoku/issues/72.

**Reply by mptrsen** [reply]
Ok, thanks! That works.

**Comment by manna-harbour** [comment]
> appreciate the info. looks like http://www.keyboard-layout-editor.com/ is broken. when I upload the json file nothing happens.

@amerikiwi93 It still works for me.  It might be a browser issue, or perhaps the file wasn't downloaded correctly?  After `Upload JSON` and choosing the `.json` file to upload, the corresponding layout should be displayed immediately.

**Reply by amerikiwi93** [reply]
I tried again and it worked this time
